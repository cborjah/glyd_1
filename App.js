import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  Dimensions,
  Image,
  KeyboardAvoidingView
} from 'react-native';

const WIDTH = Dimensions.get('window').width;

const headerText = 'Hello World!';
const imageURI =
  'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K';

export default class App extends Component {
  state = {
    showInput: false,
    inputText: '',
    text: '',
    latitude: '33.8734',
    longitude: '115.9010'
  };

  toggleInput = () => {
    this.setState({ showInput: true });
  };

  handleOnChangeText = text => this.setState({ inputText: text });

  handleOnSubmit = event => {
    this.setState({ text: event.nativeEvent.text, inputText: '' });
  };

  render() {
    const { text, latitude, longitude, showInput, inputText } = this.state;

    return (
      <View style={styles.container}>
        <KeyboardAvoidingView behavior={'position'} keyboardVerticalOffset={20}>
          {showInput ? (
            <View style={styles.contentContainer}>
              <Image source={{ uri: imageURI }} style={styles.image} />
              <Text>{this.state.text}</Text>
              <Text
                style={styles.coordinates}
              >{`(${latitude}, ${longitude})`}</Text>
            </View>
          ) : null}
          {showInput ? (
            <TextInput
              style={styles.textInput}
              onChangeText={text => this.handleOnChangeText(text)}
              value={inputText}
              onSubmitEditing={event => this.handleOnSubmit(event)}
            />
          ) : (
            <Text style={styles.header} onPress={this.toggleInput}>
              {headerText}
            </Text>
          )}
        </KeyboardAvoidingView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  contentContainer: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  header: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  },
  textInput: {
    width: WIDTH * 0.75,
    borderWidth: 0.5,
    marginTop: 20
  },
  image: {
    width: WIDTH * 0.5,
    height: WIDTH * 0.5
  }
});
