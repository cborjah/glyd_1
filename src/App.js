import React, { Component } from "react";
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  Dimensions,
  Image,
  KeyboardAvoidingView,
  TouchableOpacity,
  Share,
  Alert
} from "react-native";

const WIDTH = Dimensions.get("window").width;

const headerText = "Hello World!";
const imageURI =
  "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K";

export default class App extends Component {
  state = {
    showInput: false,
    inputText: "",
    text: "",
    latitude: null,
    longitude: null
  };

  componentDidMount() {
    this.fetchCurrentPosition();
  }

  fetchCurrentPosition = () => {
    const geoOptions = {
      timeout: 0,
      maximumAge: 0,
      enableHighAccuracy: true
    };

    // navigator.geolocation is global, no import required
    // eslint-disable-next-line no-undef
    navigator.geolocation.getCurrentPosition(
      ({ coords: { latitude, longitude } }) => {
        console.log(
          "Successfully retrieved coordinates:",
          `latitude: ${latitude}, longitude: ${longitude}`
        );
        this.setState({
          latitude: latitude.toFixed(3),
          longitude: longitude.toFixed(3)
        });
      },
      err => {
        console.log("Error retrieving location: ", err);
        Alert.alert(
          "Error",
          "Unable to retrieve your location. Please check connection and permissions."
        );
      },
      geoOptions
    );
  };

  toggleInput = () => {
    this.setState({ showInput: true });
  };

  handleOnChangeText = text => this.setState({ inputText: text });

  handleOnSubmit = event => {
    const { latitude, longitude } = this.state;

    // Retry fetching current position if needed
    if (!latitude || !longitude) {
      this.fetchCurrentPosition();
    }

    this.setState({ text: event.nativeEvent.text.trim(), inputText: "" });
  };

  handleShare = () => {
    const content = { url: imageURI };
    const options = {
      // For Email
      subject: "Check out this pic!"
    };

    Share.share(content, options);
  };

  renderImage = () => {
    return (
      <TouchableOpacity onPress={this.handleShare}>
        <Image source={{ uri: imageURI }} style={styles.image} />
      </TouchableOpacity>
    );
  };

  renderCoordinates = () => {
    const { latitude, longitude, text } = this.state;

    if (!!text.length && latitude && longitude) {
      return (
        <Text style={styles.coordinates}>{`(${latitude}, ${longitude})`}</Text>
      );
    }
  };

  render() {
    const { text, showInput, inputText } = this.state;

    return (
      <View style={[styles.container, styles.center]}>
        <KeyboardAvoidingView
          behavior="position"
          style={styles.center}
          keyboardVerticalOffset={20}
        >
          {showInput ? (
            <View style={styles.center}>
              {this.renderImage()}
              <Text style={styles.text}>{text}</Text>
              {this.renderCoordinates()}
            </View>
          ) : null}
          {showInput ? (
            <TextInput
              style={styles.textInput}
              maxLength={100}
              onChangeText={val => this.handleOnChangeText(val)}
              value={inputText}
              onSubmitEditing={event => this.handleOnSubmit(event)}
            />
          ) : (
            <TouchableOpacity onPress={this.toggleInput}>
              <Text style={styles.header}>{headerText}</Text>
            </TouchableOpacity>
          )}
        </KeyboardAvoidingView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#F5FCFF"
  },
  center: {
    justifyContent: "center",
    alignItems: "center"
  },
  header: {
    fontSize: 50,
    textAlign: "center"
  },
  text: {
    fontSize: 20,
    marginVertical: 10,
    flexWrap: "wrap"
  },
  textInput: {
    width: WIDTH * 0.75,
    borderWidth: 0.5,
    marginTop: 15,
    alignSelf: "center"
  },
  image: {
    width: WIDTH * 0.65,
    height: WIDTH * 0.65
  }
});
