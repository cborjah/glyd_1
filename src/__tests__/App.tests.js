import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import App from "../App";

describe("App", () => {
  it("renders Hello World text on mount", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("uses KeyboardAvoidingView component", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.childAt(0)).toEqual(wrapper.find("KeyboardAvoidingView"));
  });

  it("hides Hello World button after its been pressed and renders content", () => {
    const wrapper = shallow(<App />);
    const instance = wrapper.instance();

    wrapper.setState({ text: "abc", latitude: "100.000", longitude: "-50.000" });
    instance.toggleInput();

    expect(wrapper).toMatchSnapshot();
  });

  describe("TextInput's onChangeText prop", () => {
    it('sets inputText state to inputted value', () => {
      const wrapper = shallow(<App />);

      wrapper.setState({ showInput: true });
      wrapper.find("TextInput").prop("onChangeText")("abc");

      expect(wrapper.state("inputText")).toEqual("abc");
    })
  })

  describe("TextInput's onSubmitEditing prop", () => {
    it('sets text state and resets inputText state to an empty string', () => {
      const wrapper = shallow(<App />);

      wrapper.setState({ showInput: true });

      const event = { nativeEvent: { text: "abc" } };
      wrapper.find("TextInput").prop("onSubmitEditing")(event);

      expect(wrapper.state("text")).toEqual("abc");
      expect(wrapper.state("inputText")).toEqual("");
    })
  });
});
