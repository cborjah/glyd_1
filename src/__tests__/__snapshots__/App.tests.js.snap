// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App hides Hello World button after its been pressed and renders content 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <KeyboardAvoidingView
        behavior="position"
        enabled={true}
        keyboardVerticalOffset={20}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Component
              source={
                Object {
                  "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                }
              }
              style={
                Object {
                  "height": 487.5,
                  "width": 487.5,
                }
              }
            />
          </TouchableOpacity>
          <Component
            style={
              Object {
                "flexWrap": "wrap",
                "fontSize": 20,
                "marginVertical": 10,
              }
            }
          >
            abc
          </Component>
          <Component>
            (100.000, -50.000)
          </Component>
        </Component>
        <TextInput
          allowFontScaling={true}
          maxLength={100}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "borderWidth": 0.5,
              "marginTop": 15,
              "width": 562.5,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </KeyboardAvoidingView>,
      "style": Array [
        Object {
          "backgroundColor": "#F5FCFF",
          "flex": 1,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "behavior": "position",
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Component
                source={
                  Object {
                    "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                  }
                }
                style={
                  Object {
                    "height": 487.5,
                    "width": 487.5,
                  }
                }
              />
            </TouchableOpacity>
            <Component
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "marginVertical": 10,
                }
              }
            >
              abc
            </Component>
            <Component>
              (100.000, -50.000)
            </Component>
          </Component>,
          <TextInput
            allowFontScaling={true}
            maxLength={100}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "borderWidth": 0.5,
                "marginTop": 15,
                "width": 562.5,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />,
        ],
        "enabled": true,
        "keyboardVerticalOffset": 20,
        "style": Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Component
                  source={
                    Object {
                      "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                    }
                  }
                  style={
                    Object {
                      "height": 487.5,
                      "width": 487.5,
                    }
                  }
                />
              </TouchableOpacity>,
              <Component
                style={
                  Object {
                    "flexWrap": "wrap",
                    "fontSize": 20,
                    "marginVertical": 10,
                  }
                }
              >
                abc
              </Component>,
              <Component>
                (100.000, -50.000)
              </Component>,
            ],
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Component
                  source={
                    Object {
                      "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                    }
                  }
                  style={
                    Object {
                      "height": 487.5,
                      "width": 487.5,
                    }
                  }
                />,
                "onPress": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "source": Object {
                    "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                  },
                  "style": Object {
                    "height": 487.5,
                    "width": 487.5,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "abc",
                "style": Object {
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "marginVertical": 10,
                },
              },
              "ref": null,
              "rendered": "abc",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "(100.000, -50.000)",
                "style": undefined,
              },
              "ref": null,
              "rendered": "(100.000, -50.000)",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "maxLength": 100,
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "style": Object {
              "alignSelf": "center",
              "borderWidth": 0.5,
              "marginTop": 15,
              "width": 562.5,
            },
            "underlineColorAndroid": "transparent",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <KeyboardAvoidingView
          behavior="position"
          enabled={true}
          keyboardVerticalOffset={20}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Component
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Component
                source={
                  Object {
                    "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                  }
                }
                style={
                  Object {
                    "height": 487.5,
                    "width": 487.5,
                  }
                }
              />
            </TouchableOpacity>
            <Component
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "marginVertical": 10,
                }
              }
            >
              abc
            </Component>
            <Component>
              (100.000, -50.000)
            </Component>
          </Component>
          <TextInput
            allowFontScaling={true}
            maxLength={100}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "borderWidth": 0.5,
                "marginTop": 15,
                "width": 562.5,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </KeyboardAvoidingView>,
        "style": Array [
          Object {
            "backgroundColor": "#F5FCFF",
            "flex": 1,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "behavior": "position",
          "children": Array [
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Component
                  source={
                    Object {
                      "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                    }
                  }
                  style={
                    Object {
                      "height": 487.5,
                      "width": 487.5,
                    }
                  }
                />
              </TouchableOpacity>
              <Component
                style={
                  Object {
                    "flexWrap": "wrap",
                    "fontSize": 20,
                    "marginVertical": 10,
                  }
                }
              >
                abc
              </Component>
              <Component>
                (100.000, -50.000)
              </Component>
            </Component>,
            <TextInput
              allowFontScaling={true}
              maxLength={100}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "borderWidth": 0.5,
                  "marginTop": 15,
                  "width": 562.5,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />,
          ],
          "enabled": true,
          "keyboardVerticalOffset": 20,
          "style": Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                >
                  <Component
                    source={
                      Object {
                        "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                      }
                    }
                    style={
                      Object {
                        "height": 487.5,
                        "width": 487.5,
                      }
                    }
                  />
                </TouchableOpacity>,
                <Component
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 20,
                      "marginVertical": 10,
                    }
                  }
                >
                  abc
                </Component>,
                <Component>
                  (100.000, -50.000)
                </Component>,
              ],
              "style": Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeOpacity": 0.2,
                  "children": <Component
                    source={
                      Object {
                        "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                      }
                    }
                    style={
                      Object {
                        "height": 487.5,
                        "width": 487.5,
                      }
                    }
                  />,
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "source": Object {
                      "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvcR7jOw6QgU4xl7JJwTULxSZoFQ4Qj3GkDX01eQZ72ATmmO1K",
                    },
                    "style": Object {
                      "height": 487.5,
                      "width": 487.5,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "abc",
                  "style": Object {
                    "flexWrap": "wrap",
                    "fontSize": 20,
                    "marginVertical": 10,
                  },
                },
                "ref": null,
                "rendered": "abc",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "(100.000, -50.000)",
                  "style": undefined,
                },
                "ref": null,
                "rendered": "(100.000, -50.000)",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "maxLength": 100,
              "onChangeText": [Function],
              "onSubmitEditing": [Function],
              "style": Object {
                "alignSelf": "center",
                "borderWidth": 0.5,
                "marginTop": 15,
                "width": 562.5,
              },
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App renders Hello World text on mount 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 50,
              "textAlign": "center",
            }
          }
        >
          Hello World!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
